File: backend/modules/Services/CromaticaCreativa.Modules.Services.Domain/Aggregates/Service.cs
using CromaticaCreativa.Modules.Services.Domain.Enums;
using CromaticaCreativa.Modules.Services.Domain.ValueObjects;

namespace CromaticaCreativa.Modules.Services.Domain.Aggregates;

public sealed class Service
{
    private Service(
        ServiceId id,
        ServiceName name,
        string? description,
        MediaReference image,
        DisplayOrder order)
    {
        Id = id ?? throw new ArgumentNullException(nameof(id));
        Name = name ?? throw new ArgumentNullException(nameof(name));
        Description = description ?? string.Empty;
        Image = image ?? throw new ArgumentNullException(nameof(image));
        Order = order;
        Status = ServiceStatus.Inactive;
    }

    public ServiceId Id { get; }

    public ServiceName Name { get; private set; }

    public string Description { get; private set; }

    public MediaReference Image { get; private set; }

    public ServiceStatus Status { get; private set; }

    public DisplayOrder Order { get; private set; }

    public static Service Create(
        ServiceId id,
        ServiceName name,
        string? description,
        MediaReference image,
        DisplayOrder order) =>
        new(id, name, description, image, order);

    public void Rename(ServiceName name) =>
        Name = name ?? throw new ArgumentNullException(nameof(name));

    public void ChangeDescription(string? description) => Description = description ?? string.Empty;

    public void ChangeImage(MediaReference image) =>
        Image = image ?? throw new ArgumentNullException(nameof(image));

    public void ChangeOrder(DisplayOrder order) => Order = order;

    public void Activate() => Status = ServiceStatus.Active;

    public void Deactivate() => Status = ServiceStatus.Inactive;
}

File: backend/src/modules/Services/Services.Domain/Aggregates/Service.ts
import { ServiceStatus } from '../Enums/ServiceStatus';
import { DisplayOrder } from '../ValueObjects/DisplayOrder';
import { MediaReference } from '../ValueObjects/MediaReference';
import { ServiceId } from '../ValueObjects/ServiceId';
import { ServiceName } from '../ValueObjects/ServiceName';

export class Service {
  private _status = ServiceStatus.Inactive;
  private constructor(
    public readonly id: ServiceId,
    private _name: ServiceName,
    private _description: string,
    private _image: MediaReference,
    private _order: DisplayOrder,
  ) {}
  public static create(id: ServiceId, name: ServiceName, description: string | null | undefined,
    image: MediaReference, order: DisplayOrder): Service {
    return new Service(id, name, description ?? '', image, order);
  }
  public get name(): ServiceName { return this._name; }
  public get description(): string { return this._description; }
  public get image(): MediaReference { return this._image; }
  public get order(): DisplayOrder { return this._order; }
  public get status(): ServiceStatus { return this._status; }
  public rename(name: ServiceName): void { this._name = name; }
  public changeDescription(description?: string | null): void { this._description = description ?? ''; }
  public changeImage(image: MediaReference): void { this._image = image; }
  public changeOrder(order: DisplayOrder): void { this._order = order; }
  public activate(): void { this._status = ServiceStatus.Active; }
  public deactivate(): void { this._status = ServiceStatus.Inactive; }
}
